python/setup.py

```diff
@@ -28,7 +28,7 @@
 else:
     from setuptools import setup
     from setuptools.extension import Extension
-    kwargs = {'install_requires': ['numpy<=1.13.3,>=1.8.2', 'requests==2.18.4', 'graphviz==0.8.1'], 'zip_safe': False}
+    kwargs = {'install_requires': ['numpy<1.15.0,>=1.8.2', 'requests<2.19.0,>=2.18.4', 'graphviz<0.9.0,>=0.8.1'], 'zip_safe': False}
```

Comment:
Shouldn't we pin a specific version rather than specifying a range or is this the best practice in Python?

Reply:
Unlike setting up an environment where freezing all versions is preferable, we should accommodate the versions that users may have in the environment as much as we can. This change freezes the minor version, and we've tested all the minimum versions, so it's reasonable to expect newer versions not to break things.